ui/docs/reference/integrations/github/checks.mdx

```diff
@@ -0,0 +1,43 @@
+# Taskcluster-Github Checks
+
+Taskcluster-Github supports [Github's Checks API](https://developer.github.com/v3/checks/).
+At the moment, the feature is in beta. If you would like to try it out:
+* Make sure you use `.taskcluster.yml` v1
+* Add `reporting: checks-v1` line at the root level of your `.taskcluster.yml`
+* You will need `queue:route:checks` scope
+* If you are using a decision task, make sure you add `route.checks` to the `routes` of your tasks
+
+## Custom Output in Checks
+
+If you need to add some sort of custom output to your checks (for example, you would like to process
+your logs and add some results to the check's body):
+* You need to arrange for an artifact with the custom text (plain or Markdown) you'd like to add to your check 
+ (for the above example, you arrange the processing of your logs in the way you want, 
+ format the result the way you want using Markdown and save the result in an artifact)
+* Add something like 
+    ```YML
+    extra:
+        github:
+            customCheckRun:
+                textArtifactName: <artifact_name>
+    ```
+ to the definition of the task for which you want customize the check.
```

Comment:
Is there a default value for this artifact name?

Reply:
not sure what to answer (I think the answer is obvious from the code), can you explain your concern and offer suggestions?

Reply:
We had discussed having a default value in previous reviews, but I don't see it added here.  Please add one.

Reply:
but if there's a default name, how would tc-gh know if it needs to try and download the artifact? I currently do this check:

`if (customCheckRun && customCheckRun.textArtifactName)`

I'd avoid making unnecessary requests to S3 because we are billed per request. Do you have suggestions?

Reply:
The request for an artifact first goes to the queue, which doesn't bill per request, so I think it's safe to make that request in all cases.

Reply:
Oooh, I see, I see. Yes, that makes sense, thank you - I'll add the default name then!

Reply:
(do you think it's OK to _always_ hit the API tho? Because the Queue will then query the database as well....)

Reply:
Yes, it's fine

Reply:
Can you expand a bit on that and talk some more about your reasons?

Reply:
APIs are made to be called, so not calling API methods is much lower priority than user experience.  In this case, we are adding one call to `getArtifact` for each call to `createTask`, which is a much heavier-weight API call, so the marginal increase in load on queue will be very small.

That said, I had asked in an earlier review what your plans are to support images and other attributes.  Do those plans involve additional artifacts, additional properties in task.extra, and so on?  Would it make be easier for users if those were all combined into a single artifact?

Reply:
How can you combine image and text in a single artifact?

Reply:
Also, that seems like an incremental improvement to me. I don't see why we can't implement that functionality in the future when it's needed - it won't be expensive to change this code. Fenix doesn't need images right now - this PR solves their problem as it is. Just doesn't seem to be a blocking issue to me

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1624934 <---- created a bug to follow up on your suggestion

Reply:
> Just doesn't seem to be a blocking issue to me

Thinking ahead is valuable, even when it's not blocking.  If the improvement is introducing a different approach (say, a JSON artifact containing image and text), then you will need to support both the approach defined here and the new approach in parallel, until the approach defined here can be deprecated and removed.  It's possible that a small amount of thinking-ahead now can save a bunch of extra work later.  That's why I'm bringing it up now.

Reply:
As far as I know, image is binary data, so it cannot be directly inserted into JSON. So if we want to think about images and other things, we needs to think some more - which doesn't seem to be needed right now, as the need at hand does not include images.